### PR TITLE
fix: implement new API

### LINKED ISSUE #34: implement new api design
with #8 fixed, adapt the code to adhere to the api design.

### PR DESCRIPTION
**Pull request by tripodsan**
fixes #34 

BREAKING CHANGE: The Helix Admin API V1 is not compatible with the previous request handling

### implemented features:

| operation | method | path |
|-----------|--------|------|
| publish                   | `POST`   | `/live/{owner}/{repo}/{ref}/{path}` |
| preview status            | `GET`    | `/preview/{owner}/{repo}/{ref}/{path}` |
| update preview            | `POST`   | `/preview/{owner}/{repo}/{ref}/{path}` |
| preview status (reverse)  | `GET`    | `/preview/{owner}/{repo}/{ref}?editUrl={editUrl}`  |

and preview status is only partial, e.g. doesn't include last modified dates.

### CODE REVIEW

File: src/lookup/edit2web.js [hunk 1]
```diff
@@ -41,10 +41,19 @@ async function reverseLookup(context, opts) {
...
       mount,
       ...opts,
     });
+    let path = result.resourcePath;
+    if (path.endsWith('/index.md')) {
+      path = path.substring(0, path.length - 8);
+    }
+    if (path.endsWith('.md')) {
+      path = path.substring(0, path.length - 3);
+    }
```
Comment (dominique-pfister) [inline src/lookup/edit2web.js:54]: Does the performance benefit of hard-coding those numbers outweigh the potential issue of changing a string but forgetting to change its length (or length - 1) in the subsequent statement?

File: test/edit2web-google.test.js [hunk 2]
```diff
@@ -85,20 +67,12 @@ function scramble(server) {
...
+  beforeEach(() => {
     // clear cache for tests
     cache.options({ maxSize: 1000 });
-  });
-
-  beforeEach(() => {
-    fstab = FSTAB;
+    process.env.AWS_ACCESS_KEY_ID = 'fake';
+    process.env.AWS_SECRET_ACCESS_KEY = 'fake';
+    process.env.AWS_REGION = 'us-east-1';
```
Comment (dominique-pfister) [inline test/edit2web-google.test.js:75]: Doesn't `context` contain a `env` property where those variables could be specified? And why do they need to be reset before each execution?

File: src/lookup/edit2web.js (hunk 1 above)
Comment (tripodsan) [reply src/lookup/edit2web.js:54]: - the string never changes
- we can count to 8.
- the unit tests cover all cases

what wold be a better way?

like this?
```js
const SLASH_INDEX_MD = '/index.md';
const INDEX_MD = 'index.md';
if (path.endsWith(SLASH_INDEX_MD)) {
      path = path.substring(0, path.length - INDE_MD.length);    
}
```

File: test/edit2web-google.test.js (hunk 2 above)
Comment (tripodsan) [reply test/edit2web-google.test.js:75]: - no. `context` doesn't have those. they are always the ones from the environment where the lambda runs
- they could be in `before` yes.
- it seems like the aws s3 client caches the initial values and does other strange things, i.e. I noticed that:
   - if you set it to one region `A` in one test, and to a region `B` in another test, the order of tests make a difference. i.e. the 2nd test only gets the region of the first test.
   - if you don't set any secrets, the s3 client does a request to 169.168.xx.xx something and tries to set/get a token.

but I agree, that we could improve this at a later point and use the creds in the context.

File: src/lookup/edit2web.js (hunk 1 above)
Comment (dominique-pfister) [reply src/lookup/edit2web.js:54]: no, like this:

```
if (path.endsWith('/index.md')) {
  path = path.substring(0, path.length - '/index.md'.length - 1);
}
```  

In my experience, it is less error-prone to find+replace the same string in one or more locations than to adapt numbers that depend on string lengths.

File: src/lookup/edit2web.js [hunk 3]
```diff
@@ -41,10 +41,18 @@ async function reverseLookup(context, opts) {
...
   try {
     const mount = await fetchFSTab(context, opts);
-    return await handler.lookup(context, {
+    const result = await handler.lookup(context, {
       mount,
       ...opts,
     });
+    let path = result.resourcePath;
+    if (path.endsWith('/index.md')) {
+      path = path.substring(0, '/index.md'.length - 1);
```
Comment (tripodsan) [inline src/lookup/edit2web.js:50]: oh! this is wrong and the tests didn't compain!!!

Comment (tripodsan) [reply src/lookup/edit2web.js:50]: ah, they did.. I just didn't see it ;-)

**Review by dominique-pfister — APPROVED** [review]

**Review by trieloff — APPROVED** [review]

**Comment by rofe** [comment]
> BREAKING CHANGE: The Helix Admin API V1 is not compatible with the previous request handling

Does that mean the current sidekick will break? I created https://github.com/adobe/helix-sidekick/issues/12.

**Comment by tripodsan** [comment]
yes. it will break.
@stefan-guggisberg already set up a dev admin at: https://admin.helix3.dev/
so you can test it there.

**Comment by stefan-guggisberg** [comment]
for testing please use https://admin2.hlx3.page

**Pull request merged**